addons/io_hubs_addon/io/gltf_importer.py

```diff
@@ -1,342 +1,203 @@
-                    continue
-
-                else:
-                    blender_component = add_hubs_component("material", glb_component_name, glb_component_value, glb_material=glb_material)
-
-                    for property_name, property_value in glb_component_value.items():
-                        assign_property(gltf, blender_component, property_name, property_value)
-
-
-            except Exception:
-                print("Error encountered while adding Hubs components:")
-                traceback.print_exc()
-                print("Continuing on....\n")
-
-def create_scene_hubs_components(gltf):
-    if gltf.data.scene is None:
-        return
-
-    special_cases = {
-        'environment-settings': handle_environment_settings,
-        'background': handle_background,
-    }
 
-    gltf_scene = gltf.data.scenes[gltf.data.scene]
-    extensions = gltf_scene.extensions
-    if extensions:
-        MOZ_hubs_components = extensions.get('MOZ_hubs_components')
-        if MOZ_hubs_components:
-            for glb_component_name, glb_component_value in MOZ_hubs_components.items():
-                try:
-                    if glb_component_name in special_cases.keys():
-                        special_cases[glb_component_name](gltf, glb_component_name, glb_component_value)
-                        continue
-
-                    blender_component = add_hubs_component("scene", glb_component_name, glb_component_value, gltf=gltf)
-
-                    for property_name, property_value in glb_component_value.items():
-                        assign_property(gltf, blender_component, property_name, property_value)
-
-                except Exception:
-                    print("Error encountered while adding Hubs components:")
-                    traceback.print_exc()
-                    print("Continuing on....\n")
-
-
-# OBJECT SPECIAL CASES
-def handle_networked(gltf, vnode, node, glb_component_name, glb_component_value):
-    return
-
-def handle_audio_target(gltf, vnode, node, glb_component_name, glb_component_value):
-    blender_component = add_hubs_component("object", glb_component_name, glb_component_value, vnode=vnode, node=node)
-
-    for property_name, property_value in glb_component_value.items():
-        if property_name == 'srcNode':
-            setattr(blender_component, property_name, gltf.vnodes[property_value['index']].blender_object)
+def add_hubs_components(gltf_node, blender_object, import_settings):
+    if gltf_node and gltf_node.extensions and EXTENSION_NAME in gltf_node.extensions:
+        components_data = gltf_node.extensions[EXTENSION_NAME]
+        from ..components.components_registry import get_component_by_name
+        for component_name in components_data.keys():
+            component_class = get_component_by_name(component_name)
+            if component_class:
+                component_value = components_data[component_name]
+                component_class.gather_import(
```

Comment:
Sounds good, updated.

Reply:
Thank you.  However, just printing the exception that way doesn't provide any context, and so isn't very useful for debugging in cases like this.  Since this will only be printed if something goes wrong, I would like it to print the full traceback as it did in the initial version of my PR.  If you want I'll update it after merging in this PR, or you can include it now, whatever you like.